[1.16] fix shading bug of ItemMultiLayerBakedModel

cech12 opened the pull request:
Bugfix for the issue https://github.com/mezz/JustEnoughItems/issues/2087
It is a bit hacky but it works for all classes that use the `ItemMultiLayerBakedModel`. It can be undone when Forge fixed the wrong return value of the `isGui3d()` method.

cech12 committed:
fix of the shading bug of the new ItemMultiLayerBakedModel in 1.16

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/mezz/JustEnoughItems?pullRequest=2088) <br/>All committers have signed the CLA.

cech12 committed:
fix of the shading bug of the new ItemMultiLayerBakedModel in 1.16
- the isGui3d method is only used for item entities. The func_230044_c_ is called to get the "isSideLit" value which decides what shading should be used.

cech12 committed:
removed unused import

cech12 commented:
My interpretation of the `isGui3d` method was wrong. It is only used for item entities. The method which is used to decide which shading should be used in the gui is called `func_230044_c_` (later mapping: `isSideLit`). I fixed it in the last commits. 
So, I think, there is nothing wrong in the Forge implementation. You can find the vanilla call of `func_230044_c_` in `net.minecraft.client.renderer.ItemRenderer::renderItemModelIntoGUI `

cech12 committed:
add the method name of later mappings as comment

mezz commented:
Ok, that's pretty interesting. Thanks for the fix!
